### ISSUE TITLE
Optional argument of "udata licenses" is broken

### ISSUE THREAD

**Issue by pblayo** [opendatateam/udata#1270]
The second argument of `udata licenses` is broken :

    $ udata licenses https://www.data.gouv.fr/api/1/datasets/licenses
    ...
    udata: error: too many arguments

It works without the second argument because udata fetches a default list of licenses :

    $ udata licenses
    -> Dropping existing licenses
    ...
    -> Done

**Comment by pblayo** [comment]
In fact it was the udata's doc that hasn't been updated so far. See https://github.com/opendatateam/udata/pull/1273 for an update.

The flask_script's documentation isn't very clear[1]

[1] https://github.com/smurfix/flask-script/issues/143


I write a few things I've tried before a full rewrite that led me to understand the flask_script's documentation:

# Past

It never worked as specified in the documentation (with both a second argument and a default one) :

> * | e745f1d (HEAD) Improve/fix the init command
> * | 281b06c Update faker from 0.7.16 to 0.7.17

```bash
(venv) pblayo@pblayo-Latitude-E6430:~/udata$ udata licenses https://www.data.gouv.fr/api/1/datasets/licenses
usage: udata [-?]
             {clean,urls,serve,collect,purge,worker,licenses,frequency_reminder,user,search,db,generate_fixtures,init,images,cache,metrics,organizations,spatial,job,badges,api,harvest,territories,shell,runserver}
             ...
udata: error: too many arguments

(venv) pblayo@pblayo-Latitude-E6430:~/udata$ udata licenses
-> Dropping existing licenses
-> Added license "License Not Specified"
...
-> Added license "Other (Not Open)"
```

It's awkward since the code (below) is consistent with the documentation of flask_script :

```python
DEFAULT_LICENSE_FILE = 'http://licenses.opendefinition.org/licenses/groups/ckan.json'  # noqa

@manager.command
def licenses(filename=DEFAULT_LICENSE_FILE):
```

which is consistent with the [documentation](https://flask-script.readthedocs.io/en/latest/) :
> ... optional arguments:

```python
@manager.command
def hello(name="Fred")
    print "hello", name```
```

Before this change, it worked with the second argument :

> | e745f1d Improve/fix the init command
> | 281b06c (HEAD) Update faker from 0.7.16 to 0.7.17

```bash
(venv) pblayo@pblayo-Latitude-E6430:~/udata$ udata licenses https://www.data.gouv.fr/api/1/datasets/licenses
-> Dropping existing licenses
-> Added license "License Not Specified"
-> Added license "Licence Ouverte / Open Licence"
-> Added license "Open Data Commons Public Domain Dedication and Licence (PDDL)"
-> Added license "Open Data Commons Open Database License (ODbL)"
-> Added license "Open Data Commons Attribution License"
-> Added license "Creative Commons CCZero"
-> Added license "Creative Commons Attribution"
-> Added license "Creative Commons Attribution Share-Alike"
-> Added license "Other (Open)"
-> Added license "Other (Public Domain)"
-> Added license "Other (Attribution)"
-> Done

(venv) pblayo@pblayo-Latitude-E6430:~/udata$ udata licenses
usage: udata licenses [-?] filename
udata licenses: error: too few arguments
```

# Alternatives

## manager.option

I also tried (with and without `action="store_true"`):

```python
@manager.option('-l', '--licenses', dest='filename', action="store_true", default=DEFAULT_LICENSE_FILE)
def licenses(filename):
```

Which fails with :

> udata: error: too few arguments

(I also tried to replace dest by narg=1)

## Variable list of arguments

```python
@manager.command
def licenses(*args):
```

> udata: error: too many arguments

```python
@manager.command
def licenses(**kargs):
```

```python
@manager.command
def licenses(*args, **kargs):
```

    manager.add_command('licenses', licenses())

**Comment by abulte** [comment]
The `Flask-Script` doc looks pretty clear to me – once the argument becomes optional you have to specify its name, which is why the change you point to broke the legacy way to call the command:

> Or you can do optional arguments:
>
> ```
> @manager.command
> def hello(name="Fred")
>     print "hello", name
> ```
>
> These can be called like so:
>
> ```
> > python manage.py hello --name=Joe
> hello Joe
> ```

**Comment by pblayo** [comment]
Yes, exactly. I've tried different ways to keep the legacy way but it doesn't make sense and I came to the conclusion that it's the doc that need to be fixed, hence  #1273.

**Comment by pblayo** [comment]
I reopen this issue even if it's the doc that's broken, not the code (is it worth to change the title ?)
It'd be closed when #1273 is merged.
